Mathlib/CategoryTheory/Sites/NonabelianCohomology/H1.lean

```diff
@@ -0,0 +1,206 @@
+* deduce a similar result for abelian sheaves
+* when the notion of quasi-coherent sheaves on schemes is defined, show that
+if `0 ⟶ Q ⟶ M ⟶ N ⟶ 0` is an exact sequence of abelian sheaves over a scheme `X`
+and `Q` is the underlying sheaf of a quasi-coherent sheaf, then `M(U) ⟶ N(U)`
+is surjective for any affine open `U`.
+* take the colimit of `Cohomology₁ G U` over all covering families `U` (for
+a Grothendieck topology)
+
+-/
+
+universe w' w v u
+
+namespace CategoryTheory
+
+variable {C : Type u} [Category.{v} C]
+
+namespace PresheafOfGroups
+
+variable (G : Cᵒᵖ ⥤ GroupCat.{w}) {X : C} {I : Type w'} (U : I → C)
+
+/-- A zero cochain consist of a family of sections. -/
+def Cochain₀ := ∀ (i : I), G.obj (Opposite.op (U i))
+
+instance : Group (Cochain₀ G U) := by
+  dsimp [Cochain₀]
+  infer_instance
+
+namespace Cochain₀
+
+@[simp]
+lemma one_apply (i : I) : (1 : Cochain₀ G U) i = 1 := rfl
+
+@[simp]
+lemma inv_apply (γ : Cochain₀ G U) (i : I) : γ⁻¹ i = (γ i) ⁻¹ := rfl
+
+@[simp]
+lemma mul_apply (γ₁ γ₂ : Cochain₀ G U) (i : I) : (γ₁ * γ₂) i = γ₁ i * γ₂ i := rfl
+
+end Cochain₀
+
+/-- A 1-cochain of a presheaf of groups `G : Cᵒᵖ ⥤ GroupCat` on a family `U : I → C` of objects
+consists of the data of an element in `G.obj (Opposite.op T)` whenever we have elements
+`i` and `j` in `I` and maps `a : T ⟶ U i` and `b : T ⟶ U j`, and it must satisfy a compatibility
+with respect to precomposition. (When the binary product of `U i` and `U j` exists, this
+data for all `T`, `a` and `b` corresponds to the data of a section of `G` on this product.) -/
+@[ext]
+structure Cochain₁ where
+  /-- the data involved in a 1-cochain -/
+  ev (i j : I) ⦃T : C⦄ (a : T ⟶ U i) (b : T ⟶ U j) : G.obj (Opposite.op T)
+  ev_precomp (i j : I) ⦃T T' : C⦄ (φ : T ⟶ T') (a : T' ⟶ U i) (b : T' ⟶ U j) :
+    G.map φ.op (ev i j a b) = ev i j (φ ≫ a) (φ ≫ b) := by aesop
+
+namespace Cochain₁
+
+attribute [simp] Cochain₁.ev_precomp
+
+instance : One (Cochain₁ G U) where
+  one := { ev := fun _ _ _ _ _ => 1 }
+
+@[simp]
+lemma one_ev (i j : I) {T : C} (a : T ⟶ U i) (b : T ⟶ U j) :
+    (1 : Cochain₁ G U).ev i j a b = 1 := rfl
+
+variable {G U}
+
+instance : Mul (Cochain₁ G U) where
+  mul γ₁ γ₂ := { ev := fun i j T a b => γ₁.ev i j a b * γ₂.ev i j a b }
+
+@[simp]
+lemma mul_ev (γ₁ γ₂ : Cochain₁ G U) (i j : I) {T : C} (a : T ⟶ U i) (b : T ⟶ U j) :
+    (γ₁ * γ₂).ev i j a b = γ₁.ev i j a b * γ₂.ev i j a b := rfl
+
+instance : Inv (Cochain₁ G U) where
+  inv γ := { ev := fun i j T a b => (γ.ev i j a b) ⁻¹}
+
+@[simp]
+lemma inv_ev (γ : Cochain₁ G U) (i j : I) {T : C} (a : T ⟶ U i) (b : T ⟶ U j) :
+    (γ ⁻¹).ev i j a b = (γ.ev i j a b) ⁻¹ := rfl
+
+instance : Group (Cochain₁ G U) where
+  mul_assoc _ _ _ := by ext; apply mul_assoc
+  one_mul _ := by ext; apply one_mul
+  mul_one _ := by ext; apply mul_one
+  mul_left_inv _ := by ext; apply mul_left_inv
+
+end Cochain₁
+
+/-- A 1-cocycle is a 1-cochain which satisfies a cocycle condition. -/
```

Comment:
I think maybe "satisfies the cocycle condition" or "satisfies a certain cocycle condition" is better -- right now the docstring can be misinterpreted as meaning "it satisfies any kind of condition".